.github/workflows/e2e.yml

```diff
@@ -0,0 +1,77 @@
+name: E2E Tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+jobs:
+  e2e-tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        node_version:
+          - "18.19.0"
+          - "18"
+          - "20.6.0"
+          - "20"
+          - "22"
+          - "23"
+    runs-on: ubuntu-latest
+    env:
+      NPM_CONFIG_UNSAFE_PERM: true
```

Comment:
nit: I see that we're using this everywhere - I wonder if we can do without it. 🤔 can be something that can be done in a follow-up since it's not just this workflow.

Reply:
Copied from another workflow. Maybe we don't need it anymore?